spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/ContainerControllerHelperService.java

```diff
@@ -895,29 +898,24 @@ public JSONObject listSequencePartitionContainersDataTable(HttpSession session,
     try {
       JSONObject j = new JSONObject();
       JSONArray jsonArray = new JSONArray();
-      for (SequencerPartitionContainer<SequencerPoolPartition> sequencePartitionContainer : requestManager
-          .listAllSequencerPartitionContainers()) {
+      for (SequencerPartitionContainer<SequencerPoolPartition> spc : requestManager.listAllSequencerPartitionContainers()) {
         String run = "";
         String sequencer = "";
-        if (sequencePartitionContainer.getRun() != null) {
-          run = "<a href=\"/miso/run/" + sequencePartitionContainer.getRun().getId() + "\">"
-              + sequencePartitionContainer.getRun().getAlias() + "</a>";
-          if (sequencePartitionContainer.getRun().getSequencerReference() != null) {
-            sequencer = "<a href=\"/miso/sequencer/" + sequencePartitionContainer.getRun().getSequencerReference().getId() + "\">"
-                + sequencePartitionContainer.getRun().getSequencerReference().getPlatform().getNameAndModel() + "</a>";
+        if (spc.getRun() != null) {
+          run = TableHelper.hyperLinkify("/miso/run/" + spc.getRun().getId(), spc.getRun().getAlias());
+          if (spc.getRun().getSequencerReference() != null) {
+            sequencer = TableHelper.hyperLinkify("/miso/sequencer/" + spc.getRun().getSequencerReference().getId(), 
+                                     spc.getRun().getSequencerReference().getPlatform().getNameAndModel());
           }
         }
-
-        jsonArray
-            .add("['"
-                + (sequencePartitionContainer.getIdentificationBarcode() != null ? sequencePartitionContainer.getIdentificationBarcode()
-                    : "")
-                + "','"
-                + (sequencePartitionContainer.getPlatform() != null && sequencePartitionContainer.getPlatform().getPlatformType() != null
-                    ? sequencePartitionContainer.getPlatform().getPlatformType().getKey() : "")
-                + "','" + run + "','" + sequencer + "','" + "<a href=\"/miso/container/" + sequencePartitionContainer.getId()
-                + "\"><span class=\"ui-icon ui-icon-pencil\"></span></a>" + "']");
-
+        
+        JSONArray inner = new JSONArray();
+        inner.add(isStringEmptyOrNull(spc.getIdentificationBarcode()) ? TableHelper.hyperLinkify("/miso/container/" + spc.getId(), spc.getIdentificationBarcode()) : "");
+        inner.add(spc.getPlatform() != null && spc.getPlatform().getPlatformType() != null ? spc.getPlatform().getPlatformType().getKey() : "");
```

Comment:
IdentificationBarcode is not shown when present. List as unknown/"no barcode" when missing